### PR TITLE
[fuzz] Increased fuzz coverage for health check fuzz

### PR DESCRIPTION
**Pull request by zasweq**
Signed-off-by: Zach Reyes <zasweq@google.com>

Commit Message: Increased fuzz coverage for health check fuzz
Additional Description:
Increased 19 lines of coverage/357 lines total * 100 = 5.3% increase in coverage over source/common/upstream/health_checker_base_impl.cc by triggering upstream_cx_total_ in cluster stats.
Screenshot 2020-11-04 at 6 08 56 PM

Increased 12 lines of coverage/715 lines total * 100 = 1.7% increase in coverage over source/common/upstream/health_checker_impl.cc with out of range statuses (corpus entries checked in).
Risk Level: Low
Testing: Added Regression tests for increased coverage

### CODE REVIEW

**Review by adisuissa — COMMENTED** [review]
Overall looks good.
Can you explain why it was only added for HTTP and gRPC health checks and not for TCP?

File: test/common/upstream/health_check_fuzz.proto [hunk 1]
```diff
@@ -95,4 +95,5 @@ message HealthCheckTestCase {
   repeated Action actions = 2;
   bool http_verify_cluster = 3; //Determines if verify cluster setting is on
   bool start_failed = 4;
+  bool upstream_cx_total_inc = 5;
```
Comment (asraa) [inline test/common/upstream/health_check_fuzz.proto:98]: This is a really confusing title. Can you change this to "upstream_cx_success", as in, upstream connection success?

**Review by asraa — COMMENTED** [review]
Thanks!

File: test/common/upstream/health_check_fuzz.proto (hunk 1 above)
Comment (zasweq) [reply test/common/upstream/health_check_fuzz.proto:98]: Changed.

**Comment by zasweq** [comment]
Hey Adi, I based it on the unit tests, and there was nothing there for TCP in regards to upstream_cx_inc in stats.

**Comment by adisuissa** [comment]
> Hey Adi, I based it on the unit tests, and there was nothing there for TCP in regards to upstream_cx_inc in stats.

That's interesting.
I think the code that looks at `upstream_cx_total` is shared, and ivoked during [`onInterval`](https://github.com/envoyproxy/envoy/blob/5330d1d2bea34724bdc20df0ae1a8c67f2b9b22c/source/common/upstream/health_checker_base_impl.cc#L105), and that the code that [increases this counter](https://github.com/envoyproxy/envoy/blob/5330d1d2bea34724bdc20df0ae1a8c67f2b9b22c/source/common/conn_pool/conn_pool_base.cc#L467) is also being used for TCP.

**Comment by zasweq** [comment]
Okay gotcha thanks for catching that I'll add it for TCP as well :)

**Comment by adisuissa** [comment]
Thanks! Looks good.

**Review by asraa — APPROVED** [review]

**Pull request merged**